### PR TITLE
Add analytics global flag

### PR DESCRIPTION
**Pull request by delner**
This pull request adds the `Datadog.configuration.analytics_enabled` setting and `DD_TRACE_ANALYTICS_ENABLED` environment variable to globally enable and disable trace analytics.

If `true`, only integrations that aren't explicitly configured to `false` will activate analytics. If `false`, only integrations that have been explicitly configured with `true` will activate analytics. If unset (left as `nil`), only integrations that have been explicitly configured with `true` will activate analytics.

Basically this means the global flag can be used to turn on certain integrations which default as `nil` (e.g. Rack, Racecar, etc), but will not activate others (such as Excon, Sequel, etc.) Those other integrations will require explicit configuration to activate analytics. By default, all analytics are off.

### CODE REVIEW

**Comment by delner** [comment]
@brettlangdon This should be fixed, and ready for another review.
